pkg/backend/openshiftcluster.go

```diff
@@ -193,7 +193,8 @@ func (ocb *openShiftClusterBackend) updateAsyncOperation(ctx context.Context, lo
 					log.Print(backendErr)
 					asyncdoc.AsyncOperation.Error = err.CloudErrorBody
 				} else {
-					log.Error(backendErr)
+					_log := utillog.EnrichWithErrorStackTrace(log, backendErr)
+					_log.Error(backendErr)
```

Comment:
I think this is the Achilles heel.  If it were possible to make this nicer, the whole thing might fly.

The neatest options I've come up with are:
1) log.WithError(backendErr).Error(backendErr), and then pick out the Error field in a logrus hook or formatter, check if it has a backtrace and put that in a separate field
2) fork logrus.

Any others got thoughts here?

Reply:
I agree with you that this is not pretty. I will explore the first option you suggested. Hopefully we can avoid forking logrus.

Reply:
@jim-minter 
I have investigated both hook and formatter approaches. I don't think either of them will work. 
1. Hook.
Hook does not have the log object itself. The approach is designed to "hook" at the point where the log is printed and send it to somewhere else, e.g. third party tools or send to file. Theoretically we can put `log` on the hook opbject to be able to access it from within the `Fire` method, however I couldn't make this work properly - the program seems to hang when it reaches `h.log.{Print,Error,Info...}` statement. 

2. Formatter.
We can't chain Formatters so if we implement our own we lose the text formatter from logrus and we will have to duplicate its functionality. This approach will be equivalent to maintain out own fork of logrus (and tweak its text formatter)

Do we really want to go to this length to just avoid one extra line and/or have stack trace printed to the separate field? 
If we just use `log.Errorf("%v", backendErr)` we will have error text with the stacktrace in the same field. This does achieve the goal of locating the error quickly.